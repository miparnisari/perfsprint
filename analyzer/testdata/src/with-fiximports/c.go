package a

import (
	"fmt" // want "fiximports: Fix imports"
)

func loopconcat3() {
	// only fmt removed, but strings added
	s := ""
	for i := 0; i < 10; i++ {
		s += "toto" // want "concat-loop: string concatenation in a loop"
	}
	println("Hello, World!", fmt.Sprintf("%s", s)) // want "string-format: fmt.Sprintf can be replaced with just using the string"
}
