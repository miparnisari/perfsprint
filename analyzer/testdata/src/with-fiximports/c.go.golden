package a

import (
	"strings"
	// want "fiximports: Fix imports"
)

func loopconcat3() {
	// only fmt removed, but strings added
	s := ""
	var sSb10 strings.Builder
	for i := 0; i < 10; i++ {
		sSb10.WriteString("toto") // want "concat-loop: string concatenation in a loop"
	}
	s += sSb10.String()
	println("Hello, World!", s) // want "string-format: fmt.Sprintf can be replaced with just using the string"
}
